From 29dbd271a5a56983561e6113461917e00117e08f Mon Sep 17 00:00:00 2001
From: Marc Mutz <marc.mutz@qt.io>
Date: Thu, 2 Oct 2025 16:17:05 +0200
Subject: [PATCH 404/553] Mark qcborcommon_p.h as security-critical

This might be a bit controversial, but this header configures the API
of TinyCBor for use in Qt, so it's not just declarations. TinyCBor, as
third-party component, is listed as security-critical in its
qt_attribution.json, so I opted to mark the glue code as critial, too.

Amends cd4813a3c304a6d9d5d8096d91a8fc297818d624.

QUIP: 23
Task-number: QTBUG-135194
Pick-to: 6.8
Change-Id: I0042a8677eff716eb1924dfc211c740d164029cd
Reviewed-by: Matthias Rauter <matthias.rauter@qt.io>
(cherry picked from commit c9a3952c96aaecbe32d5759de3a3e9a06deee599)
Reviewed-by: Qt Cherry-pick Bot <cherrypick_bot@qt-project.org>
---
 src/corelib/serialization/qcborcommon_p.h | 1 +
 1 file changed, 1 insertion(+)

diff --git a/src/corelib/serialization/qcborcommon_p.h b/src/corelib/serialization/qcborcommon_p.h
index b80451be4bc..66c95ee887f 100644
--- a/src/corelib/serialization/qcborcommon_p.h
+++ b/src/corelib/serialization/qcborcommon_p.h
@@ -1,6 +1,7 @@
 // Copyright (C) 2018 Intel Corporation.
 // Copyright (C) 2019 The Qt Company Ltd.
 // SPDX-License-Identifier: LicenseRef-Qt-Commercial OR LGPL-3.0-only OR GPL-2.0-only OR GPL-3.0-only
+// Qt-Security score:critical reason:data-parser
 
 #ifndef QCBORCOMMON_P_H
 #define QCBORCOMMON_P_H
-- 
2.51.2

